src/CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 
-cmake_minimum_required( VERSION 3.9 )
+cmake_minimum_required( VERSION 3.23 )
```

Comment:
I guess we do not need the `if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0")` branch check on line 5?